### PR TITLE
withHost() implementation should respect PSR-7

### LINKED ISSUE #261: withHost() implementation does not respect UriInterface
According to PSR-7 UriInterface definition if an empty string passed to `withHost()` it should clear the host. The current implementation in this library automatically falls back to "localhost" in this case.

Expected:
https://github.com/php-fig/http-message/blob/master/src/UriInterface.php#L215
https://github.com/php-fig/http-message/blob/master/src/UriInterface.php#L82

Actual:
https://github.com/guzzle/psr7/blob/master/src/Uri.php#L456
https://github.com/guzzle/psr7/blob/master/src/Uri.php#L739

Using an empty space as a parameter is not a real solution for this problem, because it causes problems, for example with the string representation of the URI.

### PR DESCRIPTION
**Pull request by mxr576**
Fixes #261

### CODE REVIEW

File: tests/ServerRequestTest.php [hunk 1]
```diff
@@ -340,7 +340,7 @@ public function dataGetUriFromGlobals()
                 array_merge($server, ['REQUEST_URI' => '/blog/article.php']),
             ],
             'Empty server variable' => [
-                'http://localhost',
+                'http:',
```
Comment (mxr576) [inline tests/ServerRequestTest.php]: Not sure about this line, but I did not want to revert this commit: f3d63974d67c15a7

Comment (mxr576) [reply tests/ServerRequestTest.php]: Is this test case still required?

**Comment by sagikazarmark** [comment]
@mxr576 sorry for the delay. Can you please rebase from the latest master?

**Comment by mxr576** [comment]
PR rebased.

**Comment by mxr576** [comment]
@sagikazarmark Hi, can you take a look at this PR?

**Comment by mxr576** [comment]
@sagikazarmark ping

**Comment by Tobion** [comment]
Closing as the current behavior is neccessary to comply with RFC 7230 as the constant documents that you removed. I write in #261 how to fix the problem you encoutered.

**Pull request closed without merging**